Mathlib/Algebra/DirectSum/Ring.lean

```diff
@@ -456,16 +456,25 @@ theorem of_zero_pow (a : A 0) : ∀ n : ℕ, of A 0 (a ^ n) = of A 0 a ^ n
 instance : NatCast (A 0) :=
   ⟨GSemiring.natCast⟩
 
+
+-- TODO: These could be replaced by `map_natCast'` and `map_ofNat'` if they were
+-- marked `@[simp low]`
+-- and we added an `AddMonoidWithOne` instance added for `A 0`.
```

Comment:
Why do we need the `AddMonoidWithOne` isntance, it's implied immediately by the `Semiring` instance below.